src/databricks/labs/lakebridge/reconcile/query_builder/hash_query.py

```diff
@@ -44,12 +45,19 @@ def build_query(self, report_type: str) -> str:
         cols_with_alias = [self._build_column_with_alias(col) for col in key_cols]
 
         # in case if we have column mapping, we need to sort the target columns in the order of source columns to get
-        # same hash value
+        # same hash value. We sort by the unnormalized column name (without delimiters) to ensure consistent
+        # ordering across all dialects, regardless of their delimiter characters ([], "", ``, etc.)
+        # Fix for https://github.com/databrickslabs/lakebridge/issues/2195
         hash_cols_with_alias = [
-            {"this": self._build_column_name_source_normalized(col), "alias": self._build_alias_source_normalized(col)}
+            {
+                "this": self._build_column_name_source_normalized(col),
+                "alias": self._build_alias_source_normalized(col),
+                "sort_key": DialectUtils.unnormalize_identifier(self._build_alias_source_normalized(col)),
```

Comment:
```suggestion
                "sort_key": self._unnormalize_identifier(col),
```

this method is safer.  `DialectUtils.unnormalize_identifier` works only on ansi identifiers.